### ISSUE TITLE
visible redraws and cursor flicker in tmux splits with set title

### ISSUE THREAD

**Issue by HunterWare** [neovim/neovim#13095]
- `nvim --version`: 0.4.4 (release tarball)
- `vim -u DEFAULTS` (version: 8.2.950) behaves differently? Yes, it works properly
- Operating system/version: Centos7 and macOS Catalina
- Terminal name/version: iTerm 3.3.12 + Tmux 3.1b
- `$TERM`: screen-256color

### Steps to reproduce using `nvim -u NORC`

```
On a large terminal window, make horizontal splits (side-by-side), run nvim in any of them.

nvim -u NORC
:set title titlestring=%f\ +%l

fill the screen with newlines.
j/k up down the screen
```

### Actual behaviour

The cursor movement will be slowed, and a flickering cursor will be intermittently visible in the entire screen.  Something is visibly refreshing the entire screen every cursor movement.  The artifacts will be worse when tmux/nvim is running on the server over ssh.

### Expected behaviour

Smooth movement of cursor without slowed movement and flickering of the cursor around the entire screen.

Vim shows the expected behavior in both setups.

**Comment by justinmk** [comment]
Seems to be fixed? Test Nvim ca64b589cd74 , display behavior is as good or behavior than Vim 8.2.4113 . 

Test looked like:

<img width="1431" alt="image" src="https://user-images.githubusercontent.com/1359421/179223586-27fba60a-436e-474d-8c63-2e454a34d402.png">

**Comment by HunterWare** [comment]
Confirmed as fixed on my setup too.  Thank you!
